### DISCUSSION TITLE
找了几圈都没找到具体一点的“多用户管理“配置教程～

### DISCUSSION THREAD

**Discussion by calonye** [lobehub/lobehub#3197] [General Question]
有没有大神们指导一下，我只在开发手册看到了有关的关键变量，但是门槛有点高（https://lobehub.com/zh/docs/self-hosting/advanced/authentication#next-auth）～

有没有哪里可以翻阅具体配置步骤和相关说明的～～

想知道如何实现”多用户管理“的功能，我是本地局域网部署，期望每个人可以自行注册，登录后看到的是管理员默认配置好的一份全局参数。

**Comment by arvinxx** [comment]
参考： https://lobehub.com/zh/docs/self-hosting/advanced/server-database

**Comment by sshipanoo** [comment]
> 有没有大神们指导一下，我只在开发手册看到了有关的关键变量，但是门槛有点高（[https://lobehub.com/zh/docs/self-hosting/advanced/authentication#next-auth）～](https://lobehub.com/zh/docs/self-hosting/advanced/authentication#next-auth%EF%BC%89%EF%BD%9E)
> 
> 有没有哪里可以翻阅具体配置步骤和相关说明的～～
> 
> 想知道如何实现”多用户管理“的功能，我是本地局域网部署，期望每个人可以自行注册，登录后看到的是管理员默认配置好的一份全局参数。

我也想知道，有方案了滴我一下哎！

**Reply by shog86** [reply]
每个人可以自行注册，登录后看到的是管理员默认配置好的一份全局参数

这个目前的服务端方案做不到貌似，至于登录和注册控制可以通过Clerk侧的设置加以约束

**Reply by arvinxx** [reply]
看你要什么程度的配置了，现在有很多环境变量可以做服务端配置。比如 DEFAULT_AGENT_CONFIG 、SYSTEM_AGENT 等。

**Comment by calonye** [comment]
或者说退一步，有没有啥办法设置一份默认的参数，其他人访问的时候不需要再去设置一遍～

另外就是，我是真没找到在哪里修改 lobechat 的环境变量，我用的是 npm 编译的方式。

我之前不知道哪里看到，是可以在环境变量中配置一下参数，就能达到访问验证的方式了，也不是真的一定的账号管理的～
<img width="1012" alt="image" src="https://github.com/user-attachments/assets/4d5a17c5-f02a-454d-b87e-178476cabce6">

像图中这个，文件路径不知道在哪里。

其次参考了一下这个：
https://github.com/lobehub/lobe-chat/issues/2292
https://lobehub.com/zh/docs/self-hosting/advanced/sso-providers/auth0
---
而我只是用本地部署的 ollama，
应该是差不多这么一个格式：

-e OLLAMA_PROXY_URL=http://10.160.139.44:11434/           # ollama 访问地址
-e ENABLE_OAUTH_SSO=1   
-e NEXTAUTH_SECRET=***  # 用于加密 Auth.js 会话令牌的密钥。您可以使用以下命令： openssl rand -base64 32，或者访问 https://generate-secret.vercel.app/32 生成秘钥

-e SSO_PROVIDERS=auth0,github   #启用 Auth0 身份验证源
-e AUTH0_CLIENT_ID=**   #Auth0 应用程序的 Client ID
-e AUTH0_CLIENT_SECRET=**  #Auth0 应用程序的 Client Secret
-e AUTH0_ISSUER=https://**  #Auth0 应用程序的 Domain

-e ACCESS_CODE=123456ABCDEF  # 添加访问此服务的密码，你可以设置一个足够长的随机密码以 “禁用” 访问码授权

-e NEXTAUTH_URL=https://自己的域名/api/auth    # 该 URL 用于指定 Auth.js 在执行 OAuth 验证时的回调地址，当默认生成的重定向地址发生不正确时才需要设置
--name lobe-chat lobehub/lobe-chat


---

**Comment by King-WKR** [comment]
蹲，请问有解决这个问题吗

**Comment by ChnMig** [comment]
这个项目归根结底就是 toc 的, 目前没有特别好的方法可以做到用户管理, 模型分级 这种

**Reply by arvinxx** [reply]
对的。我们后面会有面向企业级的解决方案，但会是付费方案。LobeChat 的定位就是 toC 的个人 ai 助手，所以不会考虑在这基础上做用户管理、模型分级这些能力。

**Reply by King-WKR** [reply]
感谢，我有点不太理解，主要是我的lobechat部署在云资源上，只有一个公网IP，那我部署成功后，是不是只要有IP就能使用了。。主要是怕别人使用

**Reply by ChnMig** [reply]
> 感谢，我有点不太理解，主要是我的lobechat部署在云资源上，只有一个公网IP，那我部署成功后，是不是只要有IP就能使用了。。主要是怕别人使用

是的, 我遇到了类似的问题, 通过部署时打入变量可以锁定一部分功能(例如更改模型参数等), 但是做不到默认用户的隔离,这就回到了我说的答案, 项目设计还是toC 的, 所以他们设计时没考虑这个场景而已.
你可以看看 openwebui 或者 chatnio 吧
但是坦白说没这个好用

**Reply by arvinxx** [reply]
> 感谢，我有点不太理解，主要是我的lobechat部署在云资源上，只有一个公网IP，那我部署成功后，是不是只要有IP就能使用了。。主要是怕别人使用

你限制注册就好了呀？好多auth方案都支持不允许注册的

**Reply by ChnMig** [reply]
> > 感谢，我有点不太理解，主要是我的lobechat部署在云资源上，只有一个公网IP，那我部署成功后，是不是只要有IP就能使用了。。主要是怕别人使用
> 
> 你限制注册就好了呀？好多auth方案都支持不允许注册的

你好, 我测试auth只限制特定功能, 只有点击某些功能点才需要登录, 但是初始聊天是不限制的呀, 这也是耗费资源的. 如果设计成不登陆就无法打开页面就好了

**Reply by shog86** [reply]
我用Clerk部署的，不登录不能用

确保地址不泄露就好了

**Reply by ChnMig** [reply]
> 我用Clerk部署的，不登录不能用
> 
> 确保地址不泄露就好了

不登陆也可以使用初始的会话啊, 是我用错了吗

**Reply by shog86** [reply]
> > 我用Clerk部署的，不登录不能用
> 
> > 
> 
> > 确保地址不泄露就好了
> 
> 
> 
> 不登陆也可以使用初始的会话啊, 是我用错了吗

不登录没有部署好的模型啊，而且就算登录了也得再配置一次模型，消耗不了任何东西

**Reply by arvinxx** [reply]
> > 我用Clerk部署的，不登录不能用
> 
> > 
> 
> > 确保地址不泄露就好了
> 
> 
> 
> 不登陆也可以使用初始的会话啊, 是我用错了吗

不登录用不了的
